File: leetcode/cpp/0007_reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int neg = x < 0;
        int64_t n = 0;
        if (neg) x = -x;
        while (x > 0) {
            int a = x % 10;
            x = x / 10;
            n = n * 10 + a;
        }
        if (n >= (1LL<<31LL)) return 0;
        return neg ? -n : n;
    }
};

File: leetcode/scala/0007_reverse-integer.scala

object Solution {
    def reverse(x: Int): Int = {
        var a : Long = x
        var R : Long = 0
        
        while (a != 0) {
          var m = a % 10
          a = a / 10
          R = R * 10 + m
          if (R > Int.MaxValue || R < Int.MinValue) {
            return 0
          }
        }
        
        return R.toInt
    }
}
